test/mysql.autoupdate.test.js

```diff
@@ -788,6 +788,93 @@ describe('MySQL connector', function() {
     });
   });
 
+  it('should auto migrate/update custom foreign keys name with its constraint', function(done) {
+    var sourceModel = {
+      'name': 'SourceModel',
+      'options': {
+        'idInjection': false,
+        'mysql': {
+          'schema': 'myapp_test',
+          'table': 'source_model',
+        },
+      },
+      'properties': {
+        'id': {
+          'type': 'String',
+          'length': 20,
+          'id': 1,
+        },
+        'name': {
+          'type': 'String',
+          'length': 20,
+        },
+      },
+    };
+
+    var customFkModel = {
+      'name': 'CustomFkTest',
+      'options': {
+        'idInjection': false,
+        'mysql': {
+          'schema': 'myapp_test',
+          'table': 'custom_fk_test',
+        },
+        'foreignKeys': {
+          'fk_customfktest_customerId': {
+            'name': 'fk_customfktest_customerId',
+            'entity': 'SourceModel',
+            'entityKey': 'id',
+            'foreignKey': 'customerId',
+          },
+        },
+      },
+      'properties': {
+        'id': {
+          'type': 'String',
+          'length': 20,
+          'id': 1,
+        },
+        'customerId': {
+          'type': 'String',
+          'length': 20,
+          'mysql': {'columnName': 'custom_fk'},
+        },
+      },
+    };
+
+    var foreignKeySelect =
+    'SELECT COLUMN_NAME,CONSTRAINT_NAME,REFERENCED_TABLE_NAME, REFERENCED_COLUMN_NAME ' +
+    'FROM   INFORMATION_SCHEMA.KEY_COLUMN_USAGE ' +
+    'WHERE  REFERENCED_TABLE_SCHEMA = "myapp_test" ' +
+    'AND   TABLE_NAME = "custom_fk_test"';
+    var getCreateTable = 'SHOW CREATE TABLE `myapp_test`.`custom_fk_test`';
+
+    ds.createModel(sourceModel.name, sourceModel.properties, sourceModel.options);
+    ds.createModel(customFkModel.name, customFkModel.properties, customFkModel.options);
+
+    // do initial update/creation of table
+    ds.autoupdate(function(err) {
+      assert(!err, err);
```

Comment:
Errors should be propagated to the test runner, so that the test failure includes the original error stack trace. When using `assert`, the original stack trace is discarded and the test failure shows a stack pointing to the assertion line.

```suggestion
      if (err) return done(err);
```

Please note that we usually use async/await style in new tests.

```js
it('should auto migrate/update custom foreign keys name with its constraint', async () => {
  // ...
  const props = await ds.discoverModelProperties('custom_fk_test');
  // ...
});
```